subsys/usb/device_next/Kconfig

```diff
@@ -97,8 +97,7 @@ config USBD_MSG_WORK_DELAY
 	  yet ready to publish the message. The delay unit is milliseconds.
 
 config USBD_HWINFO_DEVID_LENGTH
-	int "The length of the device ID requested from HWINFO in bytes"
-	depends on HWINFO
+	int "The length of the device ID requested from HWINFO in bytes" if HWINFO
```

Comment:
This seems like it's just hiding the problem. Why not make `get_sn_from_hwid()` conditional to `CONFIG_HWINFO` and define `USBD_SN_ASCII7_LENGTH` accordingly?

Reply:
>This seems like it's just hiding the problem. 

What problem?

>Why not make get_sn_from_hwid() conditional to CONFIG_HWINFO and 

get_sn_from_hwid() is optimized away if HWINFO is disabled.

>define USBD_SN_ASCII7_LENGTH accordingly?

According to what?

Reply:
Could this kconfig option maybe be wrapped in `if HWINFO`? This option poses no interest to the user without HWINFO set, and `get_sn_from_hwid` can also be put behind `#ifdef HWINFO`.